tests/main/debug-execution/task.yaml

```diff
@@ -0,0 +1,163 @@
+summary: Verify snap debug execution command
+
+details: |
+    This test checks that the command `snap debug execution` shows right
+    outputs, but also verifies their correctness on the target systems
+
+debug: |
+    grep -n '' snap-*.out || true
+
+execute: |
+    snap debug execution snap > snap-default.out
+    SNAP_REEXEC=0 snap debug execution snap > snap-no-reexec.out
+    SNAP_REEXEC=1 snap debug execution snap > snap-yes-reexec.out
+
+    # this may fail when apparmor isn't found at all
+    (snap debug execution apparmor || true) > snap-apparmor-default.out
+    (SNAP_REEXEC=0 snap debug execution apparmor || true) > snap-apparmor-no-reexec.out
+
+    snap debug execution internal-tool --tool snap-update-ns > snap-uns-default.out
+
+    case "$SPREAD_SYSTEM" in
+        ubuntu-core-*)
+            echo "Checking Ubuntu Core default scenario"
+            MATCH 'distro-supports-reexec: false' < snap-default.out
+            MATCH 'is-reexec-enabled: true' < snap-default.out
+            MATCH 'is-reexec-explicitly-enabled: false' < snap-default.out
+            if os.query is-core16; then
+                # snap is part of the core snap
+                MATCH 'is-reexecd: false' < snap-default.out
+                MATCH 'self-exe: /usr/bin/snap' < snap-default.out
+            else
+                # UC18+ with snapd snap
+                MATCH 'is-reexecd: true' < snap-default.out
+                MATCH 'self-exe: /snap/snapd/.*/usr/bin/snap' < snap-default.out
+            fi
+
+            echo "Checking Ubuntu Core with reexec scenario"
+            MATCH 'distro-supports-reexec: false' < snap-yes-reexec.out
+            MATCH 'is-reexec-enabled: true' < snap-yes-reexec.out
+            MATCH 'is-reexec-explicitly-enabled: true' < snap-yes-reexec.out
```

Comment:
Maybe if those are never relevant for UC, because they are note used for anything in the code, we should not test what value they have.